### PR TITLE
changes in worker and tools related to sdk filestorage

### PR DESCRIPTION
**Pull request by muhammad-ali-e**
## What

- Implemented changes in the worker (or Runner) and tools to integrate the latest SDK feature: `Remote FileStorage`.

## Why

- To ensure compatibility between the tools and the Runner with the latest SDK, enabling seamless integration of the Remote `FileStorage` feature.

## How

- Made the Runner and tools compatible with the latest SDK by incorporating a feature flag in the Runner and conditional logic in the tools.
- In tools, used `hasattr(self, "workflow_filestorage")` for backward compatibility, as the upcoming SDK version will introduce the `workflow_filestorage` property in the `BaseTool` class.

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

- No. Backward compatibility is maintained through the feature flag and conditional checks to ensure older versions of the SDK continue to function correctly.
## Database Migrations

- 

## Env Config

- 

## Relevant Docs

-

## Related Issues or PRs

-

## Dependencies Versions

-

## Notes on Testing

-

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: tools/classifier/sample.env [hunk 1]
```diff
@@ -5,3 +5,7 @@ TOOL_DATA_DIR=../data_dir
 
 X2TEXT_HOST=http://unstract-x2text-service
 X2TEXT_PORT=3004
+
+EXECUTION_DATA_DIR=<execution_dir_path_with_bucket>
```
Comment (chandrasekharan-zipstack) [inline tools/classifier/sample.env:9]: NIT: I suggest to add a comment above these envs indicating that its for remote storage. Atleast the other 2 envs have `FS` in them but just `EXECUTION_DATA_DIR` will be confusing when we already have a lot of other similar named envs

File: tools/classifier/src/helper.py
Comment (chandrasekharan-zipstack) [inline tools/classifier/src/helper.py:1]: @muhammad-ali-e with all these tool changes, can you check and version them in `config/properties.py`? Please refer the latest versions from dockerhub - we might have missed specifying the version in the tool in between

File: tools/classifier/sample.env (hunk 1 above)
Comment (muhammad-ali-e) [reply tools/classifier/sample.env:12]: Updated env with comment. All other DIR envs will be removed once FS is implemented.

File: tools/structure/src/config/properties.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "schemaVersion": "0.0.1",
   "displayName": "Structure Tool",
   "functionName": "structure_tool",
-  "toolVersion": "0.0.46",
+  "toolVersion": "0.0.48",
```
Comment (chandrasekharan-zipstack) [inline tools/structure/src/config/properties.json:5]:
```suggestion
  "toolVersion": "0.0.50",
```

File: tools/text_extractor/src/config/properties.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
   "schemaVersion": "0.0.1",
   "displayName": "Text Extractor",
   "functionName": "text_extractor",
-  "toolVersion": "0.0.36",
+  "toolVersion": "0.0.37",
```
Comment (chandrasekharan-zipstack) [inline tools/text_extractor/src/config/properties.json:5]:
```suggestion
  "toolVersion": "0.0.38",
```

File: tools/classifier/src/config/properties.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   "schemaVersion": "0.0.1",
   "displayName": "File Classifier",
   "functionName": "classify",
-  "toolVersion": "0.0.38",
+  "toolVersion": "0.0.39",
```
Comment (chandrasekharan-zipstack) [inline tools/classifier/src/config/properties.json:5]:
```suggestion
  "toolVersion": "0.0.40",
```

**Review by chandrasekharan-zipstack — APPROVED** [review]
LGTM - @muhammad-ali-e don't you need to update the tool's requirements.txt to use the intended SDK version?

**Comment by muhammad-ali-e** [comment]
> LGTM - @muhammad-ali-e don't you need to update the tool's requirements.txt to use the intended SDK version?

@chandrasekharan-zipstack It will be updated once we transition out of the beta phase

**Review by gaya3-zipstack — APPROVED** [review]
Changes look fine. Except for tool version changes suggested by @chandrasekharan-zipstack

**Pull request merged**